include/rcpputils/split.hpp

```diff
@@ -36,34 +36,57 @@
 #ifndef RCPPUTILS__SPLIT_HPP_
 #define RCPPUTILS__SPLIT_HPP_
 
+#include <iterator>
 #include <sstream>
 #include <string>
 #include <vector>
 
 namespace rcpputils
 {
 
-/// Split a specified input into tokens using a delimiter.
+/// Split a specified input into tokens using a delimiter and a type erased insert iterator.
 /**
  * The returned vector will contain the tokens split from the input
  *
  * \param[in] input the input string to be split
  * \param[in] delim the dlelimiter used to split the input string
- * \return A vector of tokens.
+ * \param[in] insert iterator pointing to a storage contianer
  */
-inline std::vector<std::string>
-split(const std::string & input, char delim, bool skip_empty = false)
+template<
+  class InsertIterator,
+  typename std::enable_if<
```

Comment:
Could static_assert be used here instead to improve readability?

Reply:
The reason I opted for SFINAE is that `static_assert` doesn't participate in function overloading. That being, the current template approach has the benefit of having another function declaration set with `std::enable_if` which accepts a different form of iterator. One of them could be just a class member function, e.g. `add` which takes the string and operates on it.
Does this make sense?

Reply:
I see, that makes sense. I just feel it's hard to read and I wished there was a better way to write this, but I don't see any right now so this LGTM ;)